src/main/java/com/semmle/jira/addon/LgtmServlet.java

```diff
@@ -139,15 +128,32 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws I
     }
   }
 
-  void validateRequestSignature(
-      String lgtmSignature, byte[] bytes, String secret, HttpServletResponse resp)
-      throws IOException {
-    if (!Util.signatureIsValid(secret, bytes, lgtmSignature)) {
-      sendError(resp, HttpServletResponse.SC_FORBIDDEN, "Forbidden.");
+  Request validateRequest(HttpServletRequest req, String secret)
+      throws IOException, AccessControlException, InvalidRequestException {
+    byte[] bytes = IOUtils.toByteArray(req.getInputStream());
+
+    if (!Util.signatureIsValid(secret, bytes, req.getHeader("x-lgtm-signature"))) {
+      throw new AccessControlException("Forbidden");
+    }
+
+    Request request;
+    JsonNode jsonRequest;
+    try {
+      jsonRequest = Util.JSON.readTree(bytes);
+      request = new Request(jsonRequest);
+    } catch (IOException e) {
```

Comment:
I think we better catch `JsonParseException` only. Also I find the `e.getCause().getMessage()` a bit strange. Probably best to just include `e.getMessage()` as the message.